### PR TITLE
detect resource and init types and parse them

### LINKED ISSUE #6: Register possible messages while registering rules in common store

### PR DESCRIPTION
**Pull request by bhirsz**
Closes #6

### CODE REVIEW

File: robocop/run.py [hunk 1]
```diff
@@ -24,17 +25,30 @@ def __init__(self):
...
+            file_type_checker.visit(model)
+
+    def run_checks(self):
+        for file in self.files:
+            print(f"Parsing {file}")
             self.register_disablers(file)
             if self.disabler.file_disabled:
                 continue
-            model = get_model(str(file))
+            model = self.files[file].get_parser()(str(file))
```
Comment (bhirsz) [inline robocop/run.py:51]: more functional programming ;)

File: robocop/utils/file_types.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        }[self.value]
+
+
+class FileTypeChecker(ast.NodeVisitor):
+    def __init__(self, files):
+        self.files = files
+        self.source = None
+
+    def visit_ResourceImport(self, node):
+        path_normalized = node.name.replace('${/}', os.path.sep)
```
Comment (bhirsz) [inline robocop/utils/file_types.py:29]: we need to add more normalization methods, I will create issue for it (replace CURDIR etc)

File: robocop/checkers/tags.py [hunk 3]
```diff
@@ -37,5 +37,5 @@ def check_tags(self, node):
     def check_tag(self, tag, node):
         if ' ' in tag.value:
             self.report("tag-with-space", node=node, lineno=tag.lineno, col=tag.col_offset + 1)
-        if 'or' in tag.value.lower() or 'and' in tag.value.lower():
+        if 'OR' in tag.value or 'AND' in tag.value:
```
Comment (mnojek) [inline robocop/checkers/tags.py:40]: Same as in previous PR 😄 : 
```python
if any(x in ['OR', 'AND'] for x in tag.value):
```

File: robocop/run.py [hunk 4]
```diff
@@ -5,14 +5,15 @@
...
-from robocop.utils import DisablersFinder
+from robocop.utils import DisablersFinder, FileType, FileTypeChecker
 
 
 SUPPORTED_FORMATS = ('.robot')
 
 
 class Robocop:
     def __init__(self):
+        self.files = dict()
```
Comment (mnojek) [inline robocop/run.py:16]: Why not `self.files = {}`?

Comment (bhirsz) [reply robocop/run.py:16]: You're right and I'm not consistent through the same class. I will change this to {}

File: robocop/checkers/tags.py (hunk 3 above)
Comment (bhirsz) [reply robocop/checkers/tags.py:40]: In this case tag.value is string (check_tag is invoked on each tag found). 
But using your approach we can do:
```
if any(reserved in tag.value for reserved in ('OR', 'AND'):
```
Should I change it?

Comment (mnojek) [reply robocop/checkers/tags.py:40]: You're right - it's str, I didn't notice.
If we are here to play than another option is to use:
`any(map(tab.value.__contains__, ['OR', 'AND'))`
But I think that the original one is simple and more understandable.

Comment (bhirsz) [reply robocop/checkers/tags.py:40]: It's starting to look like code golf :D

**Pull request merged**